caikit/core/toolkit/name_tools.py

```diff
@@ -19,4 +19,4 @@
 
 def snake_to_upper_camel(string: str) -> str:
     """Simple snake -> upper camel conversion for descriptors"""
-    return "".join([part[0].upper() + part[1:] for part in string.split("_")])
+    return "".join([part[0].upper() + part[1:] for part in string.split("_") if part])
```

Comment:
When using `__main__` as the library, the `string.split("_")` results in `('', '', 'main', '', '')` and therefore `part[0]` raises an `IndexError` for the empty parts.